code/game/machinery/stasis.dm

```diff
@@ -116,6 +116,13 @@
 	ADD_TRAIT(target, TRAIT_TUMOR_SUPPRESSED, TRAIT_GENERIC)
 	target.extinguish_mob()
 	update_use_power(ACTIVE_POWER_USE)
+	if(ishuman(target))
+		var/mob/living/carbon/human/heart_stoppee = target
+		if(heart_stoppee.stat == DEAD || HAS_TRAIT(heart_stoppee, TRAIT_CRITICAL_CONDITION) || !heart_stoppee.can_heartattack() || (/datum/disease/heart_failure in heart_stoppee.diseases) || heart_stoppee.undergoing_cardiac_arrest())
```

Comment:
1. `set_heartattack` checks for `can_heartattack` so that check is redundant
2. Why check for `TRAIT_CRITICAL_CONDITION`? Mobs in soft crit are immune to the effects? Doesn't make much sense. 
3. `/datum/disease/heart_failure in heart_stoppee.diseases` won't work, you need to use `locate`. Also I don't see why that should be snowflaked away
4. `undergoing_cardiac_arrest` check is redundant, if they already are then `set_heartattack(TRUE)` does nothing

Reply:
i'll be honest i just stole the heart attack random event's logic

Reply:
lol

Reply:
```suggestion
		if(heart_stoppee.stat != DEAD)
			heart_stoppee.set_heartattack(TRUE)
```